sycl/doc/GettingStartedAIE.md

```diff
@@ -0,0 +1,325 @@
+make reference to in the cross-compilation commands is an Ubuntu-19.04 arm64 
+(AKA aarch64) image setup to contain all of the required libraries we need for 
+cross-compilation.
+
+In all of the following commands we assume you're running the latest
+version of Ubuntu Debian.
+
+### Compiling a basic AIE Hello World for native CPU execution
+
+The following compilation command will compile the hello_world.cpp example for 
+your native CPU. What is meant by native in this case, is whatever architecture 
+you're compiling on. So compiling on x86, will result in a hello_world 
+binary that executes on an x86 CPU. All kernels will also execute on the CPU,
+no device offloading or compilation is done.
+
+```bash
+  $SYCL_BIN_DIR/clang++ -std=c++20 hello_world.cpp -I/AIE++/acappp/include \
+    `pkg-config gtkmm-3.0 --cflags` `pkg-config gtkmm-3.0 --libs`
+```
+### Cross-compilation for ARM CPU execution
+
+The following compilation command will compile the hello_world.cpp example for 
+ARM CPU. The general idea of the compilation commands is that you're compiling 
+your binary on an Linux x86 host for an Linux ARM64 (aarch64) target.
+ 
+In this case the assumption is that the ARM target is the AIE PS which is an 
+Cortex-A72 CPU, but in theory it could be whatever `-mcpu` target the Clang 
+compiler supports. 
+ 
+```bash
+  $ISYCL_BIN_DIR/clang++ -std=c++20 -target aarch64-linux-gnu -mcpu=cortex-a72 \
+    --sysroot /net/xsjsycl41/srv/Ubuntu-19.04/arm64-root-server-rw-tmp \
+    `pkg-config gtkmm-3.0 --cflags` `pkg-config gtkmm-3.0 --libs` \
+    -I/AIE++/acappp/include  hello_world.cpp
+```
+
+### Cross-compilation for AIE PS/AI Engine execution
+
+This is the only compilation command that requires the SYCL device compiler. 
+As the intent is to compile all of the kernels for the device to be offloaded, 
+and the rest for the host.
+
+From a SYCL perspective the AIE PS is the host device that will offload to
+the other elements of the AIE architecture. The device in this case is the AIE 
+AI Engine cores.
+
+Create a script called make.sh containing
+```bash
+SYCL_TARGET=aie1_32-xilinx-unknown-sycldevice
+
+CC=$1
+
+DIRNAME=$(dirname $CC)
+INCLUDE=$(realpath $DIRNAME/../include/sycl/aie-intrinsic.h)
+
+# Edit based on environnement.
+AIE_RT_PATH=$WORKDIR/acap
+AIE_RT_BUILD_PATH=$WORKDIR/acap
+ROOT="/srv/Ubuntu-21.04/arm64-root-server-rw"
+
+shift 1
+$CC -std=c++2a -Xclang -fforce-enable-int128 \
+  -Xclang -aux-triple -Xclang aarch64-linux-gnu -g \
+  -target aarch64-linux-gnu -mcpu=cortex-a72 -fsycl -nolibsycl \
+  -DTRISYCL_XAIE_DEBUG -DTRISYCL_DEBUG -DBOOST_LOG_DYN_LINK --sysroot $ROOT \
+  -fsycl-unnamed-lambda -I$AIE_RT_BUILD_PATH_deps/experimental_mdspan-src/include/ \
+  -fsycl-targets=aie1_32-xilinx-unknown-sycldevice -Wl,-Bdynamic -ftemplate-backtrace-limit=0 \
+  -lrt -Wl,-rpath=$ROOT/usr/lib/aarch64-linux-gnu/ -fsycl-mutable-global -Xclang -fsycl-allow-func-ptr \
```

Comment:
You do not set these Clang options automatically according to `aie1_32-xilinx-unknown-sycldevice` ?